### PR TITLE
AV-23893: SDK Changes

### PR DESCRIPTION
**Pull request by chaitanyaavi**

### CODE REVIEW

File: python/avi/sdk/avi_api.py [hunk 1]
```diff
@@ -199,59 +201,55 @@ def __init__(self, controller_ip, username, password=None, token=None,
             the prefix.
         """
         super(ApiSession, self).__init__()
-        self.controller_ip = controller_ip
```
Comment (grastogi23) [inline python/avi/sdk/avi_api.py:202]: let us add property decorators for all the fields that are getting deprecated. Also, add setters for them too which will update the avi_credentials that is cached.

File: python/avi/sdk/avi_api.py [hunk 2]
```diff
@@ -179,10 +181,10 @@ class ApiSession(Session):
...
     MAX_API_RETRIES = 3
 
-    def __init__(self, controller_ip, username, password=None, token=None,
-                 tenant=None, tenant_uuid=None, verify=False, port=None,
-                 timeout=60, api_version=None,
-                 retry_conxn_errors=True, data_log=False):
+    def __init__(self, controller_ip=None, username=None, password=None,
+                 token=None, tenant=None, tenant_uuid=None, verify=False,
+                 port=None, timeout=60, api_version=None,
+                 retry_conxn_errors=True, data_log=False, avi_credentials=None):
```
Comment (grastogi23) [inline python/avi/sdk/avi_api.py:188]: add documentation that ApiSession takes ownership and may update the information inside it.

**Review by grastogi23 — CHANGES_REQUESTED** [review]

**Review by grastogi23 — APPROVED** [review]

**Comment by grastogi23** [comment]
@chaitanyaavi pl. also add test cases for the getters and setters.

**Review by nmandeyam — COMMENTED** [review]
Chaitanya, 

Can we get this into 17.1.10, 17.2.3 and eng-top.?

Narasimhan

**Review by nmandeyam — APPROVED** [review]

**Review by grastogi23 — APPROVED** [review]

**Pull request merged**